### PR TITLE
Prompt device for touch if no devices are plugged

### PR DESCRIPTION
**Pull request by codingllama**
Make sure we prompt the user if no devices are plugged, otherwise they get no feedback.

#9160

### CODE REVIEW

File: lib/auth/webauthncli/fido2.go [hunk 1]
```diff
@@ -526,16 +526,28 @@ func runOnFIDO2Devices(
...
+
+		devices, err = findSuitableDevicesOrTimeout(ctx, filter)
+	}
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	var dev deviceWithInfo
-	if shouldDoEagerPINPrompt(passwordless, devices) {
+	if !prompted && shouldDoEagerPINPrompt(passwordless, devices) {
```
Comment (codingllama) [inline lib/auth/webauthncli/fido2.go:545]: I'm asking UX about changing the prompt from "Tap your security key" to "Insert and tap your security key". If we do, then asking for the PIN right away wouldn't look so weird, so I could revert this line.

Ie, it becomes something like:

```
$ tsh login
> Insert and tap your security key
*user plugs PIN device*
> Enter your security key PIN:
(...)
```

**Comment by codingllama** [comment]
Friendly ping @nklaassen @probakowski ?

File: lib/auth/webauthncli/fido2.go [hunk 2]
```diff
@@ -526,16 +526,28 @@ func runOnFIDO2Devices(
 	prompt runPrompt, passwordless bool,
 	filter deviceFilterFunc,
 	deviceCallback deviceCallbackFunc) error {
-	devices, err := findSuitableDevicesOrTimeout(ctx, filter)
+	// Do we have readily available devices?
+	prompted := false
+	devices, err := findSuitableDevices(filter, make(map[string]struct{}))
```
Comment (nklaassen) [inline lib/auth/webauthncli/fido2.go:531]: I looked up this map param, looks like it is always passed as just an empty map, does it need to be a parameter?

**Review by nklaassen — APPROVED** [review]

File: lib/auth/webauthncli/fido2.go (hunk 2 above)
Comment (codingllama) [reply lib/auth/webauthncli/fido2.go:531]: Here it doesn't make much of a difference, but it usually outlives individual findSuitableDevices calls. I'll make everything share the same map.

**Comment by codingllama** [comment]
Thanks, Nic.

Friendly ping @probakowski ?

**Comment by codingllama** [comment]
Friendly ping @probakowski ?

**Review by probakowski — APPROVED** [review]
LGMT, sorry for the delay

**Pull request merged**